src/realm/parser/query_builder.cpp

```diff
@@ -337,6 +676,39 @@ void add_bool_constraint_to_query(Query &query, Predicate::Operator operatorType
     }
 }
 
+void add_string_constraint_to_query(Query &,
+                                    Predicate::Comparison,
+                                    SizeOperator<Size<StringData>> &&,
+                                    StringData &&) {
```

Comment:
It was like this when I inherited the code from object store so I can't comment on the original design decision, but we are in fact using temporaries for these methods and in a way it's a bit nice that if that code changes the compiler would tell you because I don't think we really want to copy around some of those expressions like `SizeOperator<Size<StringData>>`.